lib/services/auth.dart

```diff
@@ -20,17 +24,17 @@ class AuthService {
     }
   }
 
-  Future signUp(
-      String email, String password, String name, String yearId) async {
+  Future signUp(String email, String password, String name, String yearId,
+      bool isAdmin) async {
     try {
       UserCredential result = await _auth.createUserWithEmailAndPassword(
           email: email, password: password);
       User user = result.user;
-      await DatabaseService(user.uid).updateStudent(name, yearId);
+      await DatabaseService(user.uid).updateStudent(name, yearId, isAdmin);
       return user;
     } catch (error) {
       print(error.toString());
-      return null;
+      return error.toString().split("] ")[1];
```

Comment:
why you are doing something like this everywhere👀 Looks redundant in each case, but maybe u can explain

Reply:
passing an error message to let the user see why something went wrong instead of handling every possible error or exception by hand ¯\_(ツ)_/¯

Reply:
the most common error message looks like '[firestore/permission-denied] You have no access to that document'. I thought it would  be cool if I print the part without square brackets

Reply:
in registration with handling fields-oriented errors using validation, the only error from the server left is "email is already taken". That's why I used null there

Reply:
making components less huge maybe even more difficult than creating them such a huge😅 (I'm answering your comments below, idk why u placed them there)